src/Middleware/Rewrite/src/IISUrlRewrite/ConditionCollection.cs

```diff
@@ -33,7 +33,7 @@ public ConditionCollection(LogicalGrouping grouping, bool trackAllCaptures)
             {
                 return _conditions[index];
             }
-            throw new IndexOutOfRangeException($"Cannot access condition at index {index}. Only {_conditions.Count} conditions were captured.");
+            throw new ArgumentOutOfRangeException(nameof(index), $"Cannot access condition at index {index}. Only {_conditions.Count} conditions were captured.");
```

Comment:
This is another spot where the custom message is good enough that `paramName: null` might be preferred.

```suggestion
            throw new ArgumentOutOfRangeException(null, $"Cannot access condition at index {index}. Only {_conditions.Count} conditions were captured.");
```

Reply:
> how does passing in a null parameter name help?

Non-`null`/empty `paramName` is [appended to the `Message`](https://github.com/dotnet/runtime/blob/d9030cbf6a271aadd70b614b8c6e066bb1b4ca91/src/libraries/System.Private.CoreLib/src/System/ArgumentException.cs#L88-L92), which adds nothing to this message.

Reply:
I missed that you changed this before approving. Can you change it back? I think the slight redundancy in the error message is worth keeping `ArgumentOutOfRangeException.ParamName` non-null. Can we change our `ArgumentOutOfRange` exceptions to pass in the `actualValue` where it makes sense too?